packages/app-store/googlecalendar/api/webhook.ts

```diff
@@ -2,40 +2,34 @@ import type { NextApiRequest } from "next";
 
 import { HttpError } from "@calcom/lib/http-error";
 import { defaultHandler, defaultResponder } from "@calcom/lib/server";
-import prisma from "@calcom/prisma";
-import { credentialForCalendarServiceSelect } from "@calcom/prisma/selects/credential";
+import { SelectedCalendarRepository } from "@calcom/lib/server/repository/selectedCalendar";
 
 import { getCalendar } from "../../_utils/getCalendar";
 
 async function postHandler(req: NextApiRequest) {
-  // 1. validate request
-  if (req.headers["x-goog-channel-token"] !== process.env.CRON_API_KEY) {
+  if (req.headers["x-goog-channel-token"] !== process.env.GOOGLE_WEBHOOK_TOKEN) {
     throw new HttpError({ statusCode: 403, message: "Invalid API key" });
   }
   if (typeof req.headers["x-goog-channel-id"] !== "string") {
     throw new HttpError({ statusCode: 403, message: "Missing Channel ID" });
   }
 
-  const selectedCalendar = await prisma.selectedCalendar.findUnique({
-    where: {
-      googleChannelId: req.headers["x-goog-channel-id"],
-    },
-    select: {
-      credential: {
-        select: {
-          ...credentialForCalendarServiceSelect,
-          selectedCalendars: {
-            orderBy: {
-              externalId: "asc",
-            },
-          },
-        },
-      },
-    },
-  });
-  if (!selectedCalendar) throw new HttpError({ statusCode: 404, message: "No calendar found" });
+  const selectedCalendar = await SelectedCalendarRepository.findByGoogleChannelId(
+    req.headers["x-goog-channel-id"]
+  );
+
+  if (!selectedCalendar) {
+    throw new HttpError({
```

Comment:
This is weird to throw an exception for a 200. Maybe in a follow-up we can refactor this.

Reply:
I'm aware. But defaultResponder just handles the proper response and code.